src/call/mod.rs

```diff
@@ -386,12 +397,15 @@ impl Call {
     ) -> Result<BatchFuture> {
         let _cq_ref = self.cq.borrow()?;
         let i = if initial_meta { 1 } else { 0 };
-        let f = check_run(BatchType::Finish, |ctx, tag| unsafe {
+        let f = check_run_with_content(BatchType::Finish, Err(Vec::from(msg)), |content,
```

Comment:
A copy is present here, this replaces another copy in C side. We can think of a way to remove this copy to achieve real zero-copy.

IMO removing the copy needs the moving `msg`, but currently, we're **borrowing**. Any chance to change the API?